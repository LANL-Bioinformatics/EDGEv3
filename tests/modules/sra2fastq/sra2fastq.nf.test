nextflow_workflow {

    name "Test Workflow SRA2FASTQ"
    script "modules/sra2fastq/sra2fastq.nf"
    workflow "SRA2FASTQ"

    test("Basic workflow") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "PROJECT"
                    cpus = 4
                }
                sra2fastq {
                    accessions = ["SRR29462562"]
                }
            }
            workflow {
                """
                input[0] = params.sra2fastq.plus(params.shared)
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success},
                {assert snapshot(workflow.out).match()}
            )
        }

    }
    
    test("Only Download Unique Accessions") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "PROJECT"
                    cpus = 4
                }
                sra2fastq {
                    accessions = ["SRR29462562","SRR29462562","SRR29462562","SRR29462562","SRR29462562"]
                }
            }
            workflow {
                """
                input[0] = params.sra2fastq.plus(params.shared)
                """
            }
        }

        then {
            assertAll (
                {assert workflow.success},
                {assert snapshot(workflow.out).match()},
                {assert workflow.out.fastq.size() == 1}
            )
        }

    }
    test("Clean Output") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "PROJECT"
                    cpus = 4
                }
                sra2fastq {
                    accessions = ["SRR29462562"]
                    clean = true
                }
            }
            workflow {
                """
                input[0] = params.sra2fastq.plus(params.shared)
                """
            }
        }

        then {
            assertAll (
                {assert workflow.success},
                {assert snapshot(workflow.out).match()}
            )
        }

    }
}
