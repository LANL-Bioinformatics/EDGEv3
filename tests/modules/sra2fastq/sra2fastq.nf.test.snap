{
    "Only Download Unique Accessions": {
        "content": [
            {
                "0": [
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ],
                "fastq": [
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-06T14:20:55.214430761"
    },
    "Clean Output": {
        "content": [
            {
                "0": [
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ],
                "fastq": [
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-06T14:21:50.268023028"
    },
    "Basic workflow": {
        "content": [
            {
                "0": [
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ],
                "fastq": [
                    "SRR29462562.fastq.gz:md5,31506ee0f339cb6398dd326d5cb23b77"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-06T14:20:03.994730963"
    }
}